src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -401,8 +401,8 @@ seastar::future<ObjectRecoveryProgress> ReplicatedRecoveryBackend::build_push_op
     [this, &recovery_info, &progress, stat, pop]
     (auto& new_progress, auto& oi, auto& available, auto& v) {
     return [this, &recovery_info, &progress, &new_progress, &oi, pop, &v] {
+      v = recovery_info.version;
```

Comment:
could you explain the reason why we need this change in the commit message as well?

Reply:
Yeah, I'm unclear on why you need to set it after the first one.

Reply:
This is for scenarios where objects need more than one push to recover, in which later pushes will encapsulate everion 0'0 without this change. I'll add the explanation to the commit message:-)

Reply:
Ah, recovery_info.version will be eversion_t() for the first pull, so you are using v here to represent oi.version or recovery_info.version depending on whether the pull recovery_info already has the version field filled in?

Reply:
Yes, that's right